addons/sale_mrp/models/sale_mrp.py

```diff
@@ -11,19 +11,35 @@ class SaleOrderLine(models.Model):
     @api.multi
     def _compute_qty_delivered(self):
         super(SaleOrderLine, self)._compute_qty_delivered()
-
-        for line in self:
-            if line.qty_delivered_method == 'stock_move':
-                # In the case of a kit, we need to check if all components are shipped. Since the BOM might
-                # have changed, we don't compute the quantities but verify the move state.
-                bom = self.env['mrp.bom']._bom_find(product=line.product_id)
-                if bom and bom.type == 'phantom':
-                    moves = line.move_ids.filtered(lambda m: m.picking_id and m.picking_id.state != 'cancel')
-                    bom_delivered = all([move.state == 'done' for move in moves])
-                    if bom_delivered:
-                        line.qty_delivered = line.product_uom_qty
+        for order_line in self:
+            if order_line.qty_delivered_method == 'stock_move':
+                bom = self.env['mrp.bom']._bom_find(product=order_line.product_id, bom_type='phantom')
+                if bom:
```

Comment:
maybe it'd be cleaner to use the bom_id on order_line.move_ids
if it isn't set for kit products i think we could fill it anyway